developer_setup/plugins.md

```diff
@@ -0,0 +1,154 @@
+## ManageIQ Plugin Development
+
+### Plugin layout
+
+Plugins in ManageIQ are developed as [Rails Engines](http://guides.rubyonrails.org/engines.html). All models, 
+controllers etc. are loaded during boot time of ManageIQ and are just available as you would expect.
+
+### Cloning plugins
+
+It does not really matter _where_ you clone your plugins. Both approaches have their advantages.
+
+#### Side by side
+
+Having the plugins and the core repository side by side, i.e. at the same directory level, is straight forward and
+lets you separate checkouts cleanly.
+
+```bash
+cd ~/src
+git clone git@github.com:JoeSmith/manageiq-providers-amazon.git
+git clone git@github.com:JoeSmith/manageiq-content.git
+git clone git@github.com:JoeSmith/manageiq-ui-classic.git
+```
+
+#### Inside the core manageiq checkout
+
+The `plugins/` directory inside the core repository is ignored by `.gitignore`. This means you can clone plugins
+into this directory. The benefit is, e.g. when working with a project oriented IDE, you have all code under one
+directory. This way you can easily search across the core _and_ plugin code base. 
+
+You still have to change into the plugin root directory to run the tests - running `rspec` from the manageiq root will
+not pickup your plugin code, but the core code. This is a drawback which can lead to some confusion.
+
+```bash
+cd ~/src/manageiq ; mkdir plugins
+git clone git@github.com:JoeSmith/manageiq-providers-amazon.git plugins/manageiq-providers-amazon
+git clone git@github.com:JoeSmith/manageiq-content.git plugins/manageiq-content
+git clone git@github.com:JoeSmith/manageiq-ui-classic.git plugins/manageiq-ui-classic
+```
+
+### Running tests
+
+To run the tests for a plugin, we need an application context. Usually rails expects a dummy rails app. But because
+our plugins are only for manageiq, we run the tests inside the manageiq app. Therefore we need a checkout of manageiq
```

Comment:
This applies to all of the mentions of rails and manageiq elsewhere in the doc as well.